modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/ResourceManager.java

```diff
@@ -352,4 +352,101 @@ default SCIMResponse updateWithPATCHRole(String id, String patchRequest, RoleV2M
         return new SCIMResponse(ResponseCodeConstants.CODE_NOT_IMPLEMENTED, ResponseCodeConstants.DESC_NOT_IMPLEMENTED,
                 Collections.emptyMap());
     }
+
+    /**
```

Comment:
Why we define the new methods in the ResourceManager class instead of new RoleResourceV3Manager class.